### PR TITLE
Add spark2+ support

### PR DESCRIPTION
**Pull request by rmnvnv**

### CODE REVIEW

File: sparktestingbase/testcase.py [hunk 1]
```diff
@@ -14,68 +16,103 @@
...
+    def create_local_spark_session(self):
+        class_name = self.__class__.__name__
+        spark = SparkSession \
+            .builder \
+            .master("local[4]") \
+            .appName(class_name) \
+            .enableHiveSupport() \
+            .getOrCreate()
+        spark.sparkContext._conf.setAll([("spark.sql.shuffle.partitions", "12"),
+                                         ("spark.sql.session.timeZone", "UTC")])
```
Comment (angelini) [inline sparktestingbase/testcase.py]: Can we not use the config values from `get_conf` instead of repeating them here?

File: run-tests [hunk 2]
```diff
@@ -4,9 +4,17 @@ FWDIR="$(cd "`dirname $0`"; pwd)"
 cd "$FWDIR"
 
 set -x
+# Attempt to find SPARK_HOME
+if [ -z "$SPARK_HOME" ]; then
+  if hash find_spark_home.py 2>/dev/null; then
```
Comment (angelini) [inline run-tests:9]: What is this `find_spark_home.py` script, it's not described anywhere in this repo.

File: setup.py [hunk 3]
```diff
@@ -1,8 +1,8 @@
 from setuptools import setup
 
 setup(
-    name='sparktestingbase',
-    version='0.0.7-snapshot',
+    name='spark-testing-base',
```
Comment (angelini) [inline setup.py]: Why are we changing this package's name?

File: sparktestingbase/testcase.py [hunk 4]
```diff
@@ -84,11 +121,11 @@ def tearDownClass(cls):
         Tear down the basic panda spark test case. This stops the running
         context and does a hack to prevent Akka rebinding on the same port.
         """
-        cls.sc.stop()
-        derby_path = os.path.join(os.getcwd(), 'metastore_db')
+        print("stopping class")
```
Comment (angelini) [inline sparktestingbase/testcase.py]: Do we need to leave this debug logging in?

File: sparktestingbase/testcase.py [hunk 5]
```diff
@@ -14,68 +16,103 @@
...
+
+        index_expected = index_rdd(expected)
+        index_result = index_rdd(result)
+        return index_expected.cogroup(index_result) \
+            .map(lambda x: tuple(map(list, x[1]))) \
+            .filter(lambda x: x[0] != x[1]).take(1)
+
+
+class SparkTestingBaseReuse(unittest2.TestCase):
     """Basic common test case for Spark. Provides a Spark context as sc.
```
Comment (angelini) [inline sparktestingbase/testcase.py]: It is not provided as `sc`.

File: sparktestingbase/testcase.py [hunk 6]
```diff
@@ -14,68 +16,103 @@
...
+            return rdd.zipWithIndex().map(lambda x: (x[1], x[0]))
+
+        index_expected = index_rdd(expected)
+        index_result = index_rdd(result)
+        return index_expected.cogroup(index_result) \
+            .map(lambda x: tuple(map(list, x[1]))) \
+            .filter(lambda x: x[0] != x[1]).take(1)
+
+
+class SparkTestingBaseReuse(unittest2.TestCase):
```
Comment (angelini) [inline sparktestingbase/testcase.py]: Why are the new comparison methods, such as `compare_rdd` only implemented for the `BaseTestCase` and not for `BaseReuse`?

File: sparktestingbase/testcase.py (hunk 1 above)
Comment (rmnvnv) [reply sparktestingbase/testcase.py]: fixed

File: run-tests (hunk 2 above)
Comment (rmnvnv) [reply run-tests:9]: this is a "findspark" lib in setup.py. It was inherited from the source 'https://github.com/holdenk/spark-testing-base' repos

File: setup.py (hunk 3 above)
Comment (rmnvnv) [reply setup.py]: fixed

File: sparktestingbase/testcase.py (hunk 4 above)
Comment (rmnvnv) [reply sparktestingbase/testcase.py]: removed

File: sparktestingbase/testcase.py (hunk 6 above)
Comment (rmnvnv) [reply sparktestingbase/testcase.py]: I believe we don't need to have 2 classes. SparkSession builder have .getOrCreate() method and it returns current SparkSession if it exist. I removed it and leave only basic class.
Some later I'll add methods to compare dataframes.

**Review by angelini — APPROVED** [review]

**Comment by angelini** [comment]
+1

Approved with Zappr  Approved with Zappr  Approved with Zappr  Approved with Zappr  Approved with Zappr

**Pull request merged**